sentry-core/src/test/java/io/sentry/core/SendCachedEventTest.kt

```diff
@@ -0,0 +1,99 @@
+package io.sentry.core
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.whenever
+import com.nhaarman.mockitokotlin2.doThrow
+import java.io.File
+import java.nio.file.Files
+import kotlin.test.Test
+import kotlin.test.assertFalse
+
+class SendCachedEventTest {
+  private class Fixture {
+    var hub: IHub? = mock()
+      var logger: ILogger? = mock()
+      var serializer: ISerializer? = mock()
+    var options = SentryOptions()
+
+    init {
+      options.isDebug = true
+      options.setLogger(logger)
+    }
+
+    fun getSut(): SendCachedEvent {
+      return SendCachedEvent(serializer, hub, logger)
+    }
+  }
+
+    val tempDirectory = Files.createTempDirectory("send-cached-event-test")
```

Comment:
I'd declare a `File` as `lateinit`, init it using `@BeforeTest` and delete it using `@AfterTest`, so you have control if the `File` exists or not in your tests